### ISSUE TITLE
MonetDB server crashes in `vscanf`

### ISSUE THREAD

**Issue by fuboat** [MonetDB/MonetDB#7412]
**Describe the bug**
MonetDB server crashes in `vscanf`.

**To Reproduce**
```sql
SELECT CAST ( '::192.168.1.226' AS INTERVAL HOUR ( 3 ) TO SECOND );
```

**Backtrace of the Crash**
```c
#0 0x7f79b2a8af81 (vscanf+0x131)
#1 0x7f79b2a60df6 (__snprintf+0x96)
#2 0x7f79b2eb416d (parse_interval_+0x54d)
#3 0x7f79b2eb4218 (parse_interval_+0x5f8)
#4 0x7f79b2eb3a22 (parse_interval+0x612)
#5 0x7f79b2eb44d5 (interval_from_str+0x245)
#6 0x7f79b2dd8643 (second_interval_str+0x273)
#7 0x7f79b325929f (runMALsequence+0xf0f)
#8 0x7f79b325c2af (reenterMAL+0x5f)
#9 0x7f79b3489861 (OPTevaluateImplementation+0x671)
#10 0x7f79b34b97a2 (OPTwrapper+0x2c2)
#11 0x7f79b3281a7f (optimizeMALBlock+0x1cf)
#12 0x7f79b2d6bb63 (SQLoptimizeQuery+0x353)
#13 0x7f79b2d0563d (SQLparser+0x10dd)
#14 0x7f79b2d03cbc (SQLengine_+0xa5c)
#15 0x7f79b2d014fe (SQLengine+0x3e)
#16 0x7f79b327f2cf (runScenario+0x6f)
#17 0x7f79b32804de (MSscheduleClient+0xb6e)
#18 0x7f79b33b7182 (doChallenge+0x132)
#19 0x7f79b40e08a7 (thread_starter+0xc7)
#20 0x7f79b2bfb609 (start_thread+0xd9)
#21 0x7f79b2b1e133 (clone+0x43)
```

**Software versions**
 - MonetDB version number: v11.47.11 (Jun2023-SP2)
 - OS and version: Ubuntu 20.04
 - From docker image monetdb/monetdb:Jun2023-SP2

An example of steps to reproduce in docker container (**write the poc into /tmp/test.sql first**):
```bash
docker container rm monetdb_test -f                                 # remove the old one
docker run -e MDB_DB_ADMIN_PASS=monetdb --name monetdb_test -itd monetdb/monetdb:Jun2023-SP2
                                                                    # start monetdb in docker
docker exec monetdb_test bash -c 'echo -n -e "user=monetdb\npassword=monetdb\n" > /root/.monetdb'
                                                                    # set username and password
sleep 10                                                            # wait the server start
docker exec monetdb_test bash -c "grep '/usr/bin/[m]server5' /proc/*/cmdline -lR"
                                                                    # check the existence of mserver5 process
cat /tmp/test.sql | docker exec -i monetdb_test mclient monetdb     # run test case
docker exec monetdb_test bash -c "grep '/usr/bin/[m]server5' /proc/*/cmdline -lR" || { echo "[CRASH] Mserver5 process does NOT exist."; exit 1; }
                                                                    # check the existence of mserver5 process
echo "[INFO] Server normal."
```

It can also be reproduced on MonetDB Dec2023_root, self-compiled by `CC=clang-12 CXX=clang++-12 cmake /root/monetdb -DCMAKE_BUILD_TYPE=RelWithDebInfo`.

**Issue labeling**
bug

**Comment by fuboat** [comment]
I am continuously fuzzing MonetDB recently, and I plan to periodically report crashes in GitHub issues each time some new crashes are discovered. This time I found 7 new possible unique crashes in total and I am just reporting them in the GitHub issues.

Of course, if there is a more appropriate way to report these, please let me know. Thanks very much!

**Comment by yzchang** [comment]
Thank you very much for your bug reports.  Highly appreciated!

This is an excellent way to report your discoveries.  We're particularly happy with the concise queries, which allow us to easily reproduce the problems.  We'll gradually work on every problem.

**Comment by mvdvm** [comment]
Pushed a fix to the Dec2023 branch
